### DISCUSSION TITLE
Test

### DISCUSSION THREAD

**Discussion by jpwhitemn** [edgexfoundry/device-sdk-go#1175] [General]
Test of this new Discussion feature.

Looks like all repository labels are also labels for the discussion.  Assume this is ok??

Categories are the default categories.

**Comment by lenny-goodell** [comment]
That is a bummer about the labels. Would be nice to have a limited set of labels that we want for discussions. IMO, we should create a new repo just for discussions that is excluded from pushing all our standard labels.  This will give use the control we want.

**Reply by TomBrennan-3** [reply]
Is there a repo just for documentation?

**Reply by jpwhitemn** [reply]
documentation of EdgeX?  Yes - https://github.com/edgexfoundry/edgex-docs

**Comment by jpwhitemn** [comment]
@lenny-intel - that is why we see some of the project out there that are using Discussions put a special prefix in front of Discussion labels.  We can consider doing something the same.  This is also why I suggested  we set up a "dummy" repository for discussions.  That would allow labels for discussions to be independent.  Still something we can consider as we go forward.  That is why we experiment with one and then deploy to the project later.  Crawl walk run.

**Comment by farshidtz** [comment]
I think discussions per repo makes little sense even if it is just for experimentation. I'd consider the following:

- What exactly do we wanna experiment with? How much discussion is going to take place related to the SDK to validate the tools usefulness?
- How are viewers going to find the various forums? Is everyone aware of the code/functionality in each repo to ask their question or search before asking in the right place? I've been heavily involved and can't explain what half the repos are for. There is search at the organization level for discussions but I'm not sure how good it is in terms of understanding natural language.
- Where does a question related to multiple device services land?
- Where does a question about security land? Security of devices?

IMO, the use of Github Discussions with all its features is most appropriate at a higher level. I'd go ahead and say that it will be counterproductive if scoped at every repo and library. When done at a higher level, it will allow cross cutting discussions that are easily searchable. Viewers will refer to this single place to see announcements, ask questions, or find existing answers. The discussions will be visible at the organization level and easily accessible.

**Reply by farshidtz** [reply]
We could create the dummy project as proposed by @jpwhitemn and use it exclusively for device service related discussions and slowly extend it to cover other parts of the organization. This will allow us to create the appropriate categories and tags that work beyond the experimentation phase.

**Reply by lenny-goodell** [reply]
> We could create the dummy project as proposed by @jpwhitemn and use it exclusively for device service related discussions and slowly extend it to cover other parts of the organization. This will allow us to create the appropriate categories and tags that work beyond the experimentation phase.

Yes, I think this would be better so we don't have the clutter of labels currently in use for development.

**Reply by farshidtz** [reply]
@iain-anderson, I did some experiments with a dedicated project, used as the host for organizational discussions. If we call the project `edgex-discussions`:

The URL of the first discussion on the local project will be:
- https://github.com/edgexfoundry/edgex-discussions/discussions/1

The URL of the same discussion on the organization project will be:
- https://github.com/orgs/edgexfoundry/discussions/1

There will be a giant notice in the project's discussions pointing users to the organizational one. Changing the default project for the host will break the URLs since the discussion numbers are not globally unique!

So in summary, calling the project `edgex-discussions` is fine because it will not appear in the organization discussions. Hopefully, search engines will index the local project URLs.
